src/library/features/crates/cratemanager.h

```diff
@@ -0,0 +1,104 @@
+#ifndef MIXXX_CRATEMANAGER_H
+#define MIXXX_CRATEMANAGER_H
+
+#include <QSet>
+#include <QList>
+
+#include "library/crate/crate.h"
+#include "track/trackid.h"
+#include "util/db/sqlstorage.h"
+
+#include "library/crate/cratesummary.h"
+#include "library/features/crates/cratestoragehelpers.h"
+#include "library/features/crates/cratestorage.h"
+#include "library/features/crates/cratetracks.h"
+#include "library/features/crates/cratehierarchy.h"
+
+#include "util/db/fwdsqlqueryselectresult.h"
+#include "util/db/sqlsubselectmode.h"
+
+class CrateManager : public QObject, public virtual SqlStorage {
+    Q_OBJECT
+
+  public:
+    CrateManager() = default;
+    ~CrateManager() override = default;
+
+    void repairDatabase(
+      QSqlDatabase database) override;
+
+    void connectDatabase(
+            QSqlDatabase database) override;
+    void disconnectDatabase() override;
+
+    void checkClosure();
+
+    ///////////////////
+    // Member access //
+    ///////////////////
+
+    const inline CrateStorage& storage() const {
+        return m_crateStorage;
+    }
+
+    const inline CrateTracks& tracks() const {
+        return m_crateTracks;
+    }
+
+    const inline CrateHierarchy& hierarchy() const {
+        return m_crateHierarchy;
+    }
+
+    inline bool isRecursionActive() const {
```

Comment:
It is a property that changed the way a lot of stuff works, so I figured I'd put it in crateManager witch I got access to from anywhere I might need it.